security/s2a/crypter/record.go

```diff
@@ -0,0 +1,16 @@
+package crypter
+
+// S2AAeadCrypter is the interface for gRPC S2A record protocol.
+type S2AAeadCrypter interface {
+	// Encrypt encrypts the plaintext and computes the tag (if any) of dst
+	// and plaintext, dst and plaintext do not overlap.
+	Encrypt(dst, plaintext []byte) ([]byte, error)
+	// Decrypt decrypts ciphertext and verifies the tag (if any). dst and
+	// ciphertext may alias exactly or not at all. To reuse ciphertext's
+	// storage for the decrypted output, use ciphertext[:0] as dst.
```

Comment:
Can we remove the `To reuse ciphertext's...` sentence for now? We'll have to check whether all implementations of this interface actually satisfy that and if they do we can add  it back in.